client/gutenberg/extensions/business-hours/components/day.jsx

```diff
@@ -0,0 +1,210 @@
+						{ day.hours.length > 1 && (
+							<IconButton
+								isSmall
+								isLink
+								icon="trash"
+								onClick={ () => {
+									this.removeHour( hourIndex );
+								} }
+							/>
+						) }
+					</div>
+				</div>
+				{ hourIndex === day.hours.length - 1 && (
+					<div className="business-hours__row business-hours-row__add">
+						<div className={ classNames( day.name, 'business-hours__day' ) }>&nbsp;</div>
+						<div className={ classNames( day.name, 'business-hours__hours' ) }>
+							<IconButton
+								isSmall
+								isLink
+								label={ __( 'Add Hours' ) }
+								icon="plus-alt"
+								onClick={ this.addHour }
+							>
+								{ __( 'Add Hours' ) }
+							</IconButton>
+						</div>
+						<div className="business-hours__remove">&nbsp;</div>
+					</div>
+				) }
+			</Fragment>
+		);
+	};
+	setHour = ( hourValue, hourType, hourIndex ) => {
+		const { day, attributes, setAttributes } = this.props;
+		const { days } = attributes;
+		setAttributes( {
+			days: days.map( value => {
+				if ( value.name === day.name ) {
+					return {
+						...value,
+						hours: value.hours.map( ( hour, index ) => {
+							if ( index === hourIndex ) {
+								return {
+									...hour,
+									[ hourType ]: hourValue,
+								};
+							}
+							return hour;
+						} ),
+					};
+				}
+				return value;
+			} ),
+		} );
+	};
+	toggleClosed = nextValue => {
+		const { day, attributes, setAttributes } = this.props;
+		const { days } = attributes;
+
+		setAttributes( {
+			days: days.map( value => {
+				if ( value.name === day.name ) {
+					const hours = nextValue
+						? [
+								{
+									opening: defaultOpen,
+									closing: defaultClose,
+								},
+						  ]
+						: [];
+					return {
+						...value,
+						hours,
+					};
+				}
+				return value;
+			} ),
+		} );
+	};
+	addHour = () => {
+		const { day, attributes, setAttributes } = this.props;
+		const { days } = attributes;
+		day.hours.push( { opening: '', closing: '' } );
+		setAttributes( {
+			days: days.map( value => {
+				if ( value.name === day.name ) {
+					return {
+						...value,
+						hours: day.hours,
+					};
+				}
+				return value;
+			} ),
+		} );
+	};
+	removeHour = hourIndex => {
+		const { day, attributes, setAttributes } = this.props;
+		const { days } = attributes;
+
+		setAttributes( {
+			days: days.map( value => {
+				if ( day.name === value.name ) {
+					return {
+						...value,
+						hours: value.hours.filter( ( hour, index ) => {
+							return hourIndex !== index;
+						} ),
+					};
+				}
+				return value;
+			} ),
+		} );
+	};
+	isClosed() {
+		const { day } = this.props;
+		return isEmpty( day.hours );
+	}
+	renderDayToggle() {
+		const { day, edit = true, localization } = this.props;
+		return (
+			<Fragment>
+				<span className="business-hours__day-name">{ localization.days[ day.name ] }</span>
+				{ edit && (
+					<ToggleControl
+						label={ this.isClosed() ? __( 'Closed' ) : __( 'Open' ) }
```

Comment:
instead of evaluation `this.isClosed()` twice we can just do it once and store it as a value.

Reply:
no